### PR TITLE
feat(sdk/node): move js deno to node

### PR DESCRIPTION
**Pull request by afmika**
### Describe your change

Change deno frontend to pure node

### Motivation and context

Easier integration.

### Checklist

- [x] The change come with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

File: typegraph/node/dev/fix-declarations.ts [hunk 1]
```diff
@@ -6,22 +6,15 @@ import {
   fromFileUrl,
   resolve,
 } from "https://deno.land/std@0.202.0/path/mod.ts";
-import { expandGlobSync } from "https://deno.land/std@0.202.0/fs/mod.ts";
 
 export const thisDir = dirname(fromFileUrl(import.meta.url));
```
Comment (zifeo) [inline typegraph/node/dev/fix-declarations.ts:10]: with jco latest (upgrade), you can remove this entirely

File: typegraph/node/package-lock.json
Comment (zifeo) [inline typegraph/node/package-lock.json:1]: can you use ppm and try https://pnpm.io/workspaces? the reason is that with typegraph-sdk in python we can publish "multiple packages" with extras and we will like do the same with node version

File: typegraph/node/dev/fix-declarations.ts (hunk 1 above)
Comment (afmika) [reply typegraph/node/dev/fix-declarations.ts:10]: The fix-declarations.ts script?

Comment (zifeo) [reply typegraph/node/dev/fix-declarations.ts:10]: yes

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -126,10 +126,11 @@ jobs:
...
           npm install --global @bytecodealliance/jco@$JCO_VERSION
-          jco transpile $WASM_FILE -o typegraph/deno/src/gen
-          deno run -A typegraph/deno/dev/fix-declarations.ts
+          jco transpile $WASM_FILE -o typegraph/node/src/gen
 
-          cd typegraph/deno/src
+          cd typegraph/node
+          pnpm run build
+          cd dist
           deno run -A ../../../dev/publish.ts
```
Comment (zifeo) [inline .github/workflows/release.yml:134]: dev/publish.sh is not needed anymore, and we have to integrate its content into the package.json, pnpm publish should be enough for the node side

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -301,9 +301,12 @@ jobs:
           mkdir -p $(dirname $WASM_FILE)
           wasm-opt -Oz target/wasm/wasm32-unknown-unknown/debug/typegraph_core.wasm -o $WASM_FILE.opt
           wasm-tools component new $WASM_FILE.opt -o $WASM_FILE
-          rm -rf typegraph/deno/src/gen
-          jco transpile $WASM_FILE -o typegraph/deno/src/gen
-          deno run -A typegraph/deno/dev/fix-declarations.ts
+          rm -rf typegraph/node/src/gen
+          rm -rf typegraph/node/dist
```
Comment (zifeo) [inline .github/workflows/tests.yml:305]: same line?

File: typegate/tests/runtimes/python_wasi/python_wasi.ts [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
+import { PythonRuntime } from "@typegraph/sdk/runtimes/python.js";
 import outdent from "outdent";
 
 const tpe = t.struct({
   "a": t.string(),
   "b": t.list(t.either([t.integer(), t.string()])),
 });
 
-typegraph("python_wasi", (g) => {
+typegraph("python_wasi", (g: any) => {
```
Comment (zifeo) [inline typegate/tests/runtimes/python_wasi/python_wasi.ts:13]: can we type this instead?

File: typegraph/node/package.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "node",
```
Comment (zifeo) [inline typegraph/node/package.json:2]: see publish.dev to get the correct values

File: typegraph/node/src/deps/fromFileUrl.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
```
Comment (zifeo) [inline typegraph/node/src/deps/fromFileUrl.ts:2]: can you import the license header and rename the file to _import, similar to https://github.com/metatypedev/metatype/blob/main/typegate/engine/src/runtimes/prisma/engine_import.rs#L3-L17 but with the MIT License?

File: .github/workflows/release.yml (hunk 2 above)
Comment (afmika) [reply .github/workflows/release.yml:134]: fixing it

File: typegate/tests/runtimes/python_wasi/python_wasi.ts (hunk 4 above)
Comment (afmika) [reply typegate/tests/runtimes/python_wasi/python_wasi.ts:13]: I have issues on deno, it has problems typing with `*.d.ts` files when functions as arguments are involved. But on node (even pure js) the type hints works perfectly.

File: typegraph/node/package.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "@typegraph/sdk",
+  "version": "0.3.1",
```
Comment (zifeo) [inline typegraph/node/package.json:3]: likely dev/lock or dev/update needs to be adjusted to upgrade the version

File: typegraph/node/package.json [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  "description": "Declarative API development platform. Build serverless backends with zero-trust and less code, no matter where and how your (legacy) systems are.",
+  "type": "module",
+  "main": "mod.js",
```
Comment (zifeo) [inline typegraph/node/package.json:6]: can we put it inside the src folder? and rename index?

File: typegraph/node/src/deps/_import.ts [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+  let file = `${path[0]}:${path[1]}`;
+
+  if ((this as Bind)?.cb) {
+    const cb = (this as Bind).cb as any;
+    file = cb(file);
+  }
+  return file;
+}
+
+/*
```
Comment (zifeo) [inline typegraph/node/src/deps/_import.ts:56]: once per file should be enough

File: typegraph/node/src/deps/mod.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
```
Comment (zifeo) [inline typegraph/node/src/deps/mod.ts:2]: also license to import here?

**Review by zifeo — COMMENTED** [review]
almost done, what about dev/publish, I believe we can safely remove it? 

regarding the fact that the deno import using `npm:../../` does not work, can you add this a comment and a link to the upstream issue in the index.ts file?

File: examples/templates/node/tsconfig.json [hunk 11]
```diff
@@ -2,6 +2,8 @@
     "compilerOptions": {
         "strict": true,
         "moduleResolution": "node16",
-        "esModuleInterop": true
+        "target": "ES2019",
+        "module": "NodeNext",
```
[the remark this answers is not in the document]
Comment (afmika) [reply examples/templates/node/tsconfig.json:6]: Sorry it was for testing the compiled `.js` file, `module` and `target` aren't required. I'm removing them.

File: typegraph/node/package.json (hunk 7 above)
Comment (afmika) [reply typegraph/node/package.json:3]: The version and description are already automatically updated by the dev/lock.

File: typegraph/node/src/deps/mod.ts (hunk 10 above)
Comment (afmika) [reply typegraph/node/src/deps/mod.ts:2]: `mapValues` and `dirname` are custom.

File: typegraph/node/src/deps/_import.ts (hunk 9 above)
Comment (afmika) [reply typegraph/node/src/deps/_import.ts:56]: The two sections have different authors... We can merge by listing `Copyright (c) [year] [name]` line I guess.

File: typegraph/node/package.json (hunk 8 above)
Comment (afmika) [reply typegraph/node/package.json:6]: `mod.ts` is already inside src folder. But sure we can rename it!

File: typegraph/node/src/deps/_import.ts (hunk 9 above)
Comment (zifeo) [reply typegraph/node/src/deps/_import.ts:56]: yes

**Review by Yohe-Am — DISMISSED** [review]
Good work.

File: dev/lock.yml [hunk 12]
```diff
@@ -3,6 +3,7 @@ dev:
     dev/LICENSE-MPL-2.0.md:
       - typegraph/python/LICENSE.md
       - typegraph/node/LICENSE.md
```
Comment (zifeo) [inline dev/lock.yml:5]: the parent package for the workspace can be quite empty (no need for the license or other attributes) if I am not mistaken?

**Review by zifeo — APPROVED** [review]

File: dev/lock.yml (hunk 12 above)
Comment (afmika) [reply dev/lock.yml:5]: yes, `pnpm-workspace.yaml` is the minimum required file.

**Pull request merged**